lms/templates/affiliates/index.html

```diff
@@ -44,10 +53,60 @@
   </div>
 
   <script>
+    // Geographic information system (GIS) logic
+    function values (obj) {
+      return Object.keys(obj).map(function (key) {
+        return obj[key];
+      });
+    }
+
+    function buildGisApiUrl(zipcode) {
+      var baseUrl = 'https://maps.googleapis.com/maps/api/geocode/json?address=';
+
+      return baseUrl + '+' + zipcode + '+US';
+    }
+
+    function submitFormWithCoordinates(e) {
+      var $form = $('.affiliate-filter-form');
+      var zipcode = $('.affiliate-filter-form .affiliate_zipcode').val();
+      var gisApiUrl = buildGisApiUrl(zipcode);
+
+      var xhr = new XMLHttpRequest();
+      xhr.open('GET', gisApiUrl, false);
+      xhr.onload = function () {
+        if (this.status === 200) {
+          var results = JSON.parse(this.responseText).results;
+          if (results.length === 0) {
+            var latitude = 0;
+            var longitude = 0;
+          } else {
+            var location = results[0].geometry.location
+            var latitude = location.lat;
+            var longitude = location.lng;
+          }
+
+          var $latitudeField = $('<input />').attr('type', 'hidden')
```

Comment:
Instead of appending an HTML element, you could have <input type="hidden" value="" id="latitude" /> and simply update it's value before form submit.